docs/api/attachments.md

```diff
@@ -0,0 +1,176 @@
+
+## Add a new attachment:
+
+**Request**:
+
+`POST` `/attachments/`
+
+Parameters:
+
+Name            | Type   | Description
+----------------|--------|---
+filename        | string | The filename of the attachment.
+description     | string | A description of the attachment.
+s3_bucket       | string | The url of the s3_bucket where your attachment file is stored.
+s3_key          | string | The key of your s3 object.
+sequence_map_for| json   | A material object in json. The material which this attachment is a sequence map for.
+
+*Note:*
+
+- `sequence_map_for` is optional, and should only be provided when the attachment is a sequence map.
+- **[Authorization Protected](authentication.md)**
+
+**Response**:
+
+```json
+Content-Type application/json
+201 CREATED
+
+{
+   "id":18,
+   "filename":"attachment_file",
+   "description":"A file for testing",
+   "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+   "s3_key":"s3 key",
+   "deleted":"None",
+   "created_at":"2020-07-01T16:53:13+0000",
+   "updated_at":"2020-07-01T16:53:13+0000",
+   "sequence_map_for":"None"
+}
+```
+
+## Get information on an attachment:
+
+**Request**:
+
+`GET` `/attachments/:id`
+
+*Note:*
+
+- **[Authorization Protected](authentication.md)**
+- The only users who can view an attachment are:
+   1. Users who are part of an organization which has an active material request
+   2. Users who are the `requester` on an active material request
+   3. Admins
+
+**Response**:
+
+```json
+Content-Type application/json
+200 OK
+
+{
+   "id":21,
+   "filename":"attachment_file",
+   "description":"A file for testing",
+   "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+   "s3_key":"s3 key",
+   "deleted":"None",
+   "created_at":"2020-07-01T17:09:17+0000",
+   "updated_at":"2020-07-01T17:09:17+0000",
+   "sequence_map_for":"None"
+}
+```
+
+## List attachments:
+
+**Request**:
+
+`GET` `/attachments/`
+
+*Note:*
+
+- **[Authorization Protected](authentication.md)**
+- Only admins can list attachments.
+
+**Response**:
+
+```json
+Content-Type application/json
+200 OK
+
+...
+   {
+      "id":21,
+      "filename":"attachment_file",
+      "description":"A file for testing",
+      "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+      "s3_key":"s3 key",
+      "deleted":"None",
+      "created_at":"2020-07-01T17:09:17+0000",
+      "updated_at":"2020-07-01T17:09:17+0000",
+      "sequence_map_for":"None"
+   }
+...
+
+```
+
+## Update an attachment
+
+**Request**:
+
+`PUT/PATCH` `/attachments/:id`
+
+Parameters:
+
+Name            | Type   | Description
+----------------|--------|---
+filename        | string | The filename of the attachment.
+description     | string | A description of the attachment.
+s3_bucket       | string | The url of the s3_bucket where your attachment file is stored.
```

Comment:
This should just be the bucket name of the attachment file. The full URL can be constructed with the bucket and the key, but the S3 client wants them as parameters.